integration_tests/tests/idf_station.rs

```diff
@@ -126,8 +128,71 @@ async fn test_idf_station_single() {
                     metadata: IdfMetadata::new(
                         12345,
                         39,
-                        1968,
-                        2023,
+                        NaiveDate::from_ymd_opt(1968, 1, 1).unwrap(),
+                        NaiveDate::from_ymd_opt(2023, 1, 1).unwrap(),
+                        3,
+                        0,
+                        NaiveDate::from_ymd_opt(2024, 1, 1).unwrap(),
+                    ),
+                }),
+                "available station_id",
+            ),
+            (99999, None, "wrong station_id"),
+        ];
+
+        for (id, expected, case) in stations {
+            let url = format!("http://localhost:3000/reports/idf/station/{id}");
+
+            let resp = reqwest::get(url).await.expect(case);
+
+            match expected {
+                Some(expected) => {
+                    if !resp.status().is_success() {
+                        panic!("Error: {}", resp.text().await.unwrap())
+                    }
+
+                    let json: IdfStationResp = resp.json().await.expect(case);
+                    assert_eq!(json, expected, "{case}");
+                }
+                None => {
+                    // TODO: this should probably return a 404
+                    assert!(resp.status().is_server_error(), "{case}")
+                }
+            }
+        }
+    })
+    .await
+}
+
+#[tokio::test]
+async fn test_idf_station_read_file() {
+    // current directory is /integration_tests
+    let file_path = "mock_idf_station_files/mock_idf.csv";
+    let hashmap_data = parse_csv_file(file_path).unwrap();
+    write_to_csv_files("mock_idf_station_files/", hashmap_data).unwrap();
```

Comment:
This line breaks the test if you run it multiple times without deleting the written file

Reply:
True... best way to fix is that the file gets overwritten if it exists? A bit sketchy, but otherwise we have to error if it exists perhaps?